src/Adapter/Product/Update/ProductSupplierUpdater.php

```diff
@@ -63,42 +69,107 @@ class ProductSupplierUpdater
 
     /**
      * @param ProductRepository $productRepository
+     * @param CombinationRepository $combinationRepository
      * @param SupplierRepository $supplierRepository
      * @param ProductSupplierRepository $productSupplierRepository
      */
     public function __construct(
         ProductRepository $productRepository,
+        CombinationRepository $combinationRepository,
         SupplierRepository $supplierRepository,
         ProductSupplierRepository $productSupplierRepository
     ) {
         $this->productRepository = $productRepository;
         $this->supplierRepository = $supplierRepository;
         $this->productSupplierRepository = $productSupplierRepository;
+        $this->combinationRepository = $combinationRepository;
     }
 
     /**
      * @param ProductId $productId
      * @param SupplierId $defaultSupplierId
-     * @param ProductSupplier[] $productSuppliers
+     * @param array<int, ProductSupplier> $productSuppliers
      *
-     * @throws CannotUpdateProductSupplierException
-     * @throws CoreException
-     * @throws ProductSupplierNotFoundException
+     * @return array<int, ProductSupplierId>
      */
-    public function setProductSuppliers(ProductId $productId, SupplierId $defaultSupplierId, array $productSuppliers): void
-    {
-        $deletableProductSupplierIds = $this->getDeletableProductSupplierIds($productId->getValue(), $productSuppliers);
+    public function setProductSuppliers(
+        ProductId $productId,
+        SupplierId $defaultSupplierId,
+        array $productSuppliers
+    ): array {
+        $product = $this->productRepository->get($productId);
 
-        foreach ($productSuppliers as $productSupplier) {
-            if ($productSupplier->id) {
-                $this->productSupplierRepository->update($productSupplier);
-            } else {
-                $this->productSupplierRepository->add($productSupplier);
-            }
+        if ($product->cache_default_attribute) {
```

Comment:
I think it may be safer to use `$product->hasCombinations()` just in case `cache_default_attribute` isn't consistent I know it implies an extra query but it seems safer

I'm not blocking the PR for this though, so I leave the decision to you 😉